src/core/ddsc/src/dds_reader.c

```diff
@@ -665,8 +669,15 @@ static dds_entity_t dds_create_reader_int (dds_entity_t participant_or_subscribe
     iox_sub_storage_extension_init(&rd->m_iox_sub_stor);
 
     assert (rqos->durability.kind == DDS_DURABILITY_VOLATILE);
-    opts.queueCapacity = rd->m_entity.m_domain->gv.config.sub_queue_capacity;
-    opts.historyRequest = 0;
+    // opts.queueCapacity = rd->m_entity.m_domain->gv.config.sub_queue_capacity;
+    opts.queueCapacity = (uint64_t)rqos->history.depth;
+
+    // MAKI: if this is always zero we need no
+    // capacity on the writer side ...
+    // opts.historyRequest = 0;
+    // TODO: will those mechanism used by iceoryx and cyclonedds to get data
+    // at late joining readers interfere?
+    opts.historyRequest = (uint64_t)rqos->history.depth;
```

Comment:
A late joiner only needs (or: is allowed to get, depending on how you interpret things) historical data if its durability kind ≠ `VOLATILE`. So I think it should be 0 for volatile readers.

As an aside: in DDS the amount of historical data available is determined by the writer (for `TRANSIENT_LOCAL`) or the topic (for `TRANSIENT` and `PERSISTENT` — those are NIY so don't matter), and for `TRANSIENT_LOCAL` at least, the reader is force-fed with more than anything. Naturally, you end up with at most `rqos->history.depth` samples/instance, so using this for an Iceoryx history request seems reasonable.

And also beware of the `assert (rqos->durability.kind == DDS_DURABILITY_VOLATILE)`.

Reply:
Allowing non-volatile readers (e.g. extending to transient local) is not part of this pull request since there are other issues to be resolved first.

I wonder how useful this assert was anyway, if it has to be ensured at all times that we need a volatile reader. As of now, nothing is checked in this regard in a release build.

Reply:
Changed the logic on reader and writer side to essentially not care anymore for those parameters. It was pointless before to have iceoryx buffer data at the writer side (history capacity) but never requesting this data anyway in the volatile case (sub history request = 0).

Now those are always zero for the volatile reader case (the only case currently supported, but this is to be extended in another issue #965). However, we no limit the subscriber queue size to the history depth. This saves memory under load but leads to data being lost earlier than before (still within spec). If we do not want this we should always set it to the maximum available in iceoryx or leave it to the configuration as it was.
Note that the history request and history capacity still make no sense as they are used and need to be removed or the semantics changed.